src/main/scala/swiss/dasch/api/AuthService.scala

```diff
@@ -5,27 +5,38 @@
 
 package swiss.dasch.api
 
+import swiss.dasch.domain.AuthScope
 import pdi.jwt.*
 import swiss.dasch.config.Configuration.JwtConfig
 import zio.*
+import zio.json.*
 import zio.prelude.Validation
+import swiss.dasch.api.AuthService.JwtContents
 
 trait AuthService {
-  def authenticate(jwtToken: String): ZIO[Any, NonEmptyChunk[AuthenticationError], JwtClaim]
+  def authenticate(jwtToken: String): ZIO[Any, NonEmptyChunk[AuthenticationError], (AuthScope, JwtClaim)]
 }
 
 object AuthService {
-  def authenticate(token: String): ZIO[AuthService, NonEmptyChunk[AuthenticationError], JwtClaim] =
+  def authenticate(token: String): ZIO[AuthService, NonEmptyChunk[AuthenticationError], (AuthScope, JwtClaim)] =
```

Comment:
question (not-sure-myself): since you're creating domain objects from the JWT here anyway, would it make sense to do that all the way, and not return a `pdi.jwt.JwtClaim` any more at all, and instead model all the needed information as domain objects?  
Admittedly, I do not know off the top of my head, where and how that is used

Reply:
Maybe it's better if it'd return a UserSession in the first place? BaseEndpoints is unpacking some obscure tuple, AuthService might as well return the whole tuple. :)

Reply:
I think we should have a `Principal` returned by the `AuthService`:

`final case class Principal(subject: String, scope: AuthScope)`

The `subject` being the `sub` claim value from the JWT. 

I find a naming like `User[Session|Profile]` misleading because in certain cases we might not have a real life _User_ as the subject of the JWT. An access token might have been issued for a system rather than a user. For example when importing/exporting projects the subject is the dsp-api and not a user. Should we ever use the dsp-ingest with some scheduled automation (e.g. clean up tasks) then a real life user is also not present.

Reply:
+1 For the `JwtClaim` should not be exposed by the `AuthService`.

This is implementation detail and authorization logic should only depend on the principal.